main.tf

```diff
@@ -337,25 +355,56 @@ data "aws_iam_policy_document" "lb_log_delivery" {
       values   = ["bucket-owner-full-control"]
     }
   }
+}
 
-  statement {
-    sid = "AWSLogDeliveryAclCheck"
+# Cloudtrail
+data "aws_caller_identity" "this" {}
 
-    effect = "Allow"
+data "aws_iam_policy_document" "cloudtrail_log_delivery" {
+  count = var.create_bucket && var.attach_cloudtrail_log_delivery_policy ? 1 : 0
+
+  statement {
+    sid = "AWSCloudTrailAclCheck"
 
     principals {
       type        = "Service"
-      identifiers = ["delivery.logs.amazonaws.com"]
+      identifiers = ["cloudtrail.amazonaws.com"]
     }
 
+    effect = "Allow"
+
     actions = [
       "s3:GetBucketAcl",
     ]
 
     resources = [
       aws_s3_bucket.this[0].arn,
     ]
+  }
+
+  statement {
+    sid = "AWSCloudTrailWrite"
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
 
+    actions = [
+      "s3:PutObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.this[0].arn}/AWSLogs/${data.aws_caller_identity.this.account_id}/*",
```

Comment:
There can be `optionalPrefix` in ARN.

Also, the policy is not exactly matching the one described in the [AWS docs](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/create-s3-bucket-policy-for-cloudtrail.html). e.g. missing `aws:SourceArn` in conditions.

I am not sure if it is going to work with the one you have or not. Please double-check.

Reply:
No problem, I will add this ;)

Reply:
I limit the rights on the bucket to the "most strict", as indicated on the documentation (specify path in bucket for write).
For ALB/ELB/NLB/LB logs, there is normally also a way to make the rights stricter, like [here](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html).
However, this is not what is currently implemented in the module.
Do I follow what is currently there (less strict) or do I propose to modify the others to be more strict as well?

Reply:
So I propose a "security" improvement ;)